componentapiexample/src/main/java/net/gini/android/vision/component/camera/BaseCameraScreenHandler.java

```diff
@@ -0,0 +1,357 @@
+                Toast.LENGTH_LONG).show();
+    }
+
+    private void launchReviewScreen(final Document document) {
+        final Intent intent = getReviewActivityIntent(document);
+        mActivity.startActivityForResult(intent, REVIEW_REQUEST);
+    }
+
+    protected abstract Intent getReviewActivityIntent(final Document document);
+
+    private void launchAnalysisScreen(final Document document) {
+        final Intent intent = getAnalysisActivityIntent(document);
+        mActivity.startActivityForResult(intent, ANALYSIS_REQUEST);
+    }
+
+    protected abstract Intent getAnalysisActivityIntent(final Document document);
+
+    protected Activity getActivity() {
+        return mActivity;
+    }
+
+    protected GiniVisionFeatureConfiguration getGiniVisionFeatureConfiguration() {
+        return mGiniVisionFeatureConfiguration;
+    }
+
+    public void onActivityResult(final int requestCode, final int resultCode, final Intent data) {
+        switch (requestCode) {
+            case REVIEW_REQUEST:
+            case ANALYSIS_REQUEST:
+                if (resultCode == Activity.RESULT_OK) {
+                    mActivity.finish();
+                }
+                break;
+        }
+    }
+
+    public boolean onBackPressed() {
+        if (isOnboardingVisible()) {
+            removeOnboarding();
+            return true;
+        }
+        return false;
+    }
+
+    protected abstract boolean isOnboardingVisible();
+
+    public void onCreate(final Bundle savedInstanceState) {
+        setUpActionBar();
+        setTitlesForCamera();
+
+        configureLogging();
+        setupGiniVisionCoordinator(mActivity);
+
+        // Configure the features you would like to use
+        mGiniVisionFeatureConfiguration =
+                GiniVisionFeatureConfiguration.buildNewConfiguration()
+                        .setDocumentImportEnabledFileTypes(
+                                DocumentImportEnabledFileTypes.PDF_AND_IMAGES)
+                        .setFileImportEnabled(true)
+                        .build();
+
+        if (savedInstanceState == null) {
+            final Intent intent = mActivity.getIntent();
+            if (isIntentActionViewOrSend(intent)) {
+                startGiniVisionLibraryForImportedFile(intent);
+            } else {
+                showCamera();
+            }
+        } else {
+            mCameraFragmentInterface = retainCameraFragment();
+        }
+    }
+
+    protected abstract void setUpActionBar();
+
+    private void showCamera() {
+        LOG.debug("Show the Camera Screen");
+        mCameraFragmentInterface = showCameraFragment();
+        // Delay notifying the coordinator to allow the camera fragment view to be created
+        new Handler().post(new Runnable() {
+            @Override
+            public void run() {
+                mGiniVisionCoordinator.onCameraStarted();
+            }
+        });
+    }
+
+    protected abstract CameraFragmentInterface showCameraFragment();
```

Comment:
Does not the method name represent the action? Maybe instead of _show_ it could be _get_ or _retrieve_.
I say this because 10 lines above you are assigning it to a variable, I know that you are returning the fragment after showing it but maybe it can be confusing.